### PR TITLE
switch to http server for deployment

### PR DESCRIPTION
**Pull request by camallen**
PR to simplify the local docker-compose setup and startup of the app for container runtimes. 

Specifically to:
- allow port to be configurable for dev & prod
- run server on http
- use docker CMD start script with ENV var configuration opts for erlang and elixir
- avoid ssl setup on boot
- allow docker compose to run the tests via the built image

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use Mix.Config
 # watchers to your application. For example, we use it
 # with brunch.io to recompile .js and .css sources.
 config :designator, Designator.Endpoint,
-  http: [port: 4000],
+  http: [port: {:system, "PORT"}],
```
Comment (adammcmaster) [inline config/dev.exs]: What happens if `PORT` isn't defined? Can we add a default?

File: start.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# ensure we stop on error (-e) and log cmds (-x)
+set -ex
+ERLANG_NODE_NAME=${ERLANG_NODE_NAME:-erlang@designator.zooniverse.org}
+ERLANG_DISTRIBUTED_PORT=${ERLANG_DISTRIBUTED_PORT:-9001}
+# enable the SMP (multiprocessing) for erlang VM, https://docs.riak.com/riak/kv/latest/using/performance/erlang/index.html#smp
+ERLANG_SMP=${ERLANG_SMP:-auto}
+ERLANG_KERNEL_OPTS=${ERLANG_KERNEL_OPTS:-"inet_dist_listen_min ${ERLANG_DISTRIBUTED_PORT} inet_dist_listen_max ${ERLANG_DISTRIBUTED_PORT}"}
+ERLANG_OPTS="-smp ${ERLANG_SMP} -kernel ${ERLANG_KERNEL_OPTS}"
+# run the elixir app via Erlang VM
+elixir --name $ERLANG_NODE_NAME --cookie $SECRET_KEY_BASE --erl "${ERLANG_OPTS}" -S mix phoenix.server
```
Comment (adammcmaster) [inline start.sh]:
```suggestion
exec elixir --name $ERLANG_NODE_NAME --cookie $SECRET_KEY_BASE --erl "${ERLANG_OPTS}" -S mix phoenix.server
```

File: config/dev.exs (hunk 1 above)
Comment (camallen) [reply config/dev.exs]: looks like that's not supported as this is a runtime vs build time issue, https://elixirforum.com/t/what-is-the-difference-between-using-system-port-and-system-get-env-port-in-deployment/1975/47

I can add a sensible default in the start script?

Comment (camallen) [reply config/dev.exs]: added

**Pull request merged**